Duplicate messages from multiple jobs when using tox.ini config

### Steps to reproduce
```ini
##############################################################################
# PYLINT SETTINGS                                                            #
##############################################################################
[MASTER]
# Use multiple processes to speed up Pylint.
jobs=2
```
```bash
ls package3
__init__.py module.py   module2.py  module3.py
```
module.py
```python
import os


class Test:
    def do_1(self):
        pass

    def do_2(self):
        pass

    def do_3(self):
        pass

    def do_4(self):
        pass

    def do_5(self):
        pass

    def do_6(self):
        pass

    def do_7(self):
        pass

    def do_8(self):
        pass

    def do_9(self):
        pass
```
module2.py
```python
import os


class Test:
    def do(other):
        print('here')

```
module3.py
```python
import os

```
### Current behavior
produce duplicate of error for each job
```bash
pylint --rcfile=tox.ini package3
************* Module package3.module2
package3/module2.py:5:4: E0213: Method should have "self" as first argument (no-self-argument)
package3/module2.py:5:4: E0213: Method should have "self" as first argument (no-self-argument)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 8.86/10, -0.86)
```
### Expected behavior
no duplicates
```bash
pylint --rcfile=tox.ini package3
************* Module package3.module2
package3/module2.py:5:4: E0213: Method should have "self" as first argument (no-self-argument)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 8.86/10, -0.86)
```
### pylint --version output
```
pylint 2.5.3
astroid 2.4.2
Python 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

this issue is really annoying

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -403,3 +403,5 @@ contributors:
 * Yeting Li (yetingli): contributor
 
 * Frost Ming (frostming): contributor
+
+* Eli Fine (eli88fine): Fixed false positive duplicate code warning for lines with symbols only

--- a/ChangeLog
+++ b/ChangeLog
@@ -23,6 +23,8 @@ Release date: TBA
 
 * Fix a crash when a specified config file does not exist
 
+* Fix ``duplicate-code`` false positive when lines only contain whitespace and non-alphanumeric characters (e.g. parentheses, bracket, comman, etc.)
+
 What's New in Pylint 2.6.0?
 ===========================
 

--- a/pylint/checkers/similar.py
+++ b/pylint/checkers/similar.py
@@ -14,6 +14,7 @@
 # Copyright (c) 2019 Taewon D. Kim <kimt33@mcmaster.ca>
 # Copyright (c) 2019 Pierre Sassoulas <pierre.sassoulas@gmail.com>
 # Copyright (c) 2020 Shiv Venkatasubrahmanyam <shvenkat@users.noreply.github.com>
+# Copyright (c) 2020 Eli Fine <ejfine@gmail.com>
 
 # Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 # For details: https://github.com/PyCQA/pylint/blob/master/COPYING
@@ -22,6 +23,7 @@
 """a similarities / code duplication command line tool and pylint checker
 """
 
+import re
 import sys
 from collections import defaultdict
 from getopt import getopt
@@ -129,21 +131,21 @@ class Similar:
             skip = 1
             num = 0
             for index2 in find(lineset1[index1]):
-                non_blank = 0
+                num_lines_with_content = 0
                 for num, ((_, line1), (_, line2)) in enumerate(
                     zip(lines1(index1), lines2(index2))
                 ):
                     if line1 != line2:
-                        if non_blank > min_lines:
+                        if num_lines_with_content > min_lines:
                             yield num, lineset1, index1, lineset2, index2
                         skip = max(skip, num)
                         break
-                    if line1:
-                        non_blank += 1
+                    if re.match(r"\w", line1):
+                        num_lines_with_content += 1
                 else:
                     # we may have reach the end
                     num += 1
-                    if non_blank > min_lines:
+                    if num_lines_with_content > min_lines:
                         yield num, lineset1, index1, lineset2, index2
                     skip = max(skip, num)
             index1 += skip

--- a/pylint/checkers/similar.py
+++ b/pylint/checkers/similar.py
@@ -36,6 +36,8 @@ from pylint.interfaces import IRawChecker
 from pylint.reporters.ureports.nodes import Table
 from pylint.utils import decoding_stream
 
+REGEX_FOR_LINES_WITH_CONTENT = re.compile(r".*\w+")
+
 
 class Similar:
     """finds copy-pasted lines of code in a project"""
@@ -140,10 +142,10 @@ class Similar:
                             yield num, lineset1, index1, lineset2, index2
                         skip = max(skip, num)
                         break
-                    if re.match(r"\w", line1):
+                    if re.match(REGEX_FOR_LINES_WITH_CONTENT, line1):
                         num_lines_with_content += 1
                 else:
-                    # we may have reach the end
+                    # we may have reached the end
                     num += 1
                     if num_lines_with_content > min_lines:
                         yield num, lineset1, index1, lineset2, index2

--- a/tests/checkers/unittest_similar.py
+++ b/tests/checkers/unittest_similar.py
@@ -9,6 +9,7 @@
 # Copyright (c) 2019-2020 Pierre Sassoulas <pierre.sassoulas@gmail.com>
 # Copyright (c) 2019 Ashley Whetter <ashley@awhetter.co.uk>
 # Copyright (c) 2019 Taewon D. Kim <kimt33@mcmaster.ca>
+# Copyright (c) 2020 Eli Fine <ejfine@gmail.com>
 
 # Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 # For details: https://github.com/PyCQA/pylint/blob/master/COPYING
@@ -24,6 +25,8 @@ from pylint.checkers import similar
 INPUT = Path(__file__).parent / ".." / "input"
 SIMILAR1 = str(INPUT / "similar1")
 SIMILAR2 = str(INPUT / "similar2")
+SIMILAR3 = str(INPUT / "similar3")
+SIMILAR4 = str(INPUT / "similar4")
 MULTILINE = str(INPUT / "multiline-import")
 HIDE_CODE_WITH_IMPORTS = str(INPUT / "hide_code_with_imports.py")
 
@@ -175,6 +178,19 @@ TOTAL lines=60 duplicates=5 percent=8.33
 """
             % (SIMILAR1, SIMILAR2)
         ).strip()
+    )
+
+
+def test_lines_without_meaningful_content_do_not_trigger_similarity():
+    output = StringIO()
+    with redirect_stdout(output), pytest.raises(SystemExit) as ex:
+        similar.Run([SIMILAR3, SIMILAR4])
+    assert ex.value.code == 0
+    assert (
+        output.getvalue().strip()
+        == """
+TOTAL lines=18 duplicates=0 percent=0.00
+""".strip()
     )
 
 

--- a/tests/checkers/unittest_similar.py
+++ b/tests/checkers/unittest_similar.py
@@ -188,9 +188,29 @@ def test_lines_without_meaningful_content_do_not_trigger_similarity():
     assert ex.value.code == 0
     assert (
         output.getvalue().strip()
-        == """
-TOTAL lines=18 duplicates=0 percent=0.00
-""".strip()
+        == (
+            """
+14 similar lines in 2 files
+==%s:11
+==%s:11
+   b = (
+       (
+           [
+               "Lines 12-25 still trigger a similarity...",
+               "...warning, because..."
+           ],
+           [
+               "...even after ignoring lines with only symbols..."
+           ],
+       ),
+       (
+           "...there are still 5 similar lines in this code block.",
+       )
+   )
+TOTAL lines=50 duplicates=14 percent=28.00
+"""
+            % (SIMILAR3, SIMILAR4)
+        ).strip()
     )
 
 

--- a/tests/input/similar3
+++ b/tests/input/similar3
@@ -1,0 +1,9 @@
+one
+two
+three
+four = (
+    [
+    ],
+    [
+    ]
+)

--- a/tests/input/similar3
+++ b/tests/input/similar3
@@ -1,9 +1,25 @@
-one
-two
-three
-four = (
-    [
-    ],
-    [
-    ]
+a = (
+    (
+        [
+            "Lines 1-8 used to trigger a similarity warning, but now they don't because...",
+            "...lines with only symbols are ignored and don't count in the line count limit."
+        ]
+    ),
+)
+
+"This line in similar3 breaks up the sections of code by being different than similar4"
+
+b = (
+    (
+        [
+            "Lines 12-25 still trigger a similarity...",
+            "...warning, because..."
+        ],
+        [
+            "...even after ignoring lines with only symbols..."
+        ],
+    ),
+    (
+        "...there are still 5 similar lines in this code block.",
+    )
 )

--- a/tests/input/similar4
+++ b/tests/input/similar4
@@ -1,9 +1,25 @@
-one
-two
-three
-four = (
-    [
-    ],
-    [
-    ]
+a = (
+    (
+        [
+            "Lines 1-8 used to trigger a similarity warning, but now they don't because...",
+            "...lines with only symbols are ignored and don't count in the line count limit."
+        ]
+    ),
+)
+
+"This line in similar4 breaks up the sections of code by being different than similar3"
+
+b = (
+    (
+        [
+            "Lines 12-25 still trigger a similarity...",
+            "...warning, because..."
+        ],
+        [
+            "...even after ignoring lines with only symbols..."
+        ],
+    ),
+    (
+        "...there are still 5 similar lines in this code block.",
+    )
 )

--- a/tests/input/similar4
+++ b/tests/input/similar4
@@ -1,0 +1,9 @@
+one
+two
+three
+four = (
+    [
+    ],
+    [
+    ]
+)

